### ISSUE TITLE
Embed Kepler without Redux store

### ISSUE THREAD

**Issue by ericsoco** [keplergl/kepler.gl#1170]
Our design is such that we'd like to isolate Kepler from the rest of the application, and do not want to expose our store to Kepler. It appears that Kepler's use of `connect` prohibits use of Kepler without a Redux store.

I was hoping to use `keplerGlReducer` within a `useReducer` call, but it looks like I'll need to create a second Redux store (via `createStore`) and pass that in as Kepler's `store` prop.

I'm not clear enough yet on Kepler's internals to understand what a refactor to enable using Kepler without an external store would look like, but presumably it would entail Kepler creating its own store.

Any ideas welcome! And may be able to issue a PR to make this happen given guidance.

**Comment by chrisgervang** [comment]
Since a store is just an object, isolated to itself, I wonder if there is a way to use the react-redux Provider around where you want to start isolating it from the rest of your app. Can a Provider be used twice to override a parent Provider?

**Comment by ericsoco** [comment]
@chrisgervang sure, but I still need to provide a store, whether I do so via Context (`<Provider>`) or props (`store` prop, which is undocumented as part of Kepler's API but passed into `connect()` by default, along with other props.

**Comment by ericsoco** [comment]
FWIW, looks like [Superset sets up a Redux store just for Kepler](https://github.com/apache/incubator-superset/pull/6403/files#diff-98a82f6fe725c1656fc9f985cefcf45bR61) (and [shims in `react-palm`](https://docs.kepler.gl/docs/api-reference/get-started#1-mount-reducer), which is mentioned as a req in Kepler's docs).

Still, if Kepler team is interested in Kepler optionally providing its own store, I can try cutting a PR in a couple weeks.

**Comment by heshan0131** [comment]
@ericsoco as long as it's an optional feature, feel free to start a PR on it

**Comment by chrisgervang** [comment]
I’m making an assumption that your application uses redux too, and I’m not sure we’re on the same page. To rephrase, could your application maintain 2 redux stores to keep things isolated? 

Perhaps the outer Provider context can be flushed so an _inner isolated_ Provider context can be provided just for the Kepler component.

**Comment by ericsoco** [comment]
@chrisgervang yes, my application uses Redux (it's a Fusion.js app), and yes, I can maintain two stores. That's exactly what I'm doing now.

However, I'm finding I'm unable to dispatch actions to Kepler (getting `Reducers may not dispatch actions.` errors on `dispatch()`), presumably because Kepler is connecting to the wrong context. I could specify a [custom context](https://react-redux.js.org/api/hooks#custom-context) on my Provider, but this is of no use given that I can't tell Kepler to connect to this context.

I didn't intend for this ticket to be a bug report, more of a feature request, but I've spent pretty much all day trying to stand up Kepler in my app, to no avail. If the host app could avoid jumping through all these Redux hoops and instead just provide a store to Kepler directly, and tie Kepler's reducers back into the host app's store, things would be a lot simpler.

Note similar issues users have filed in #351. FWIW, my app is on `react-redux@7.2.0` (after upgrading to match Kepler's version and pinning Kepler's via `yarn resolutions`, as I found (ht @Firenze11, via [this comment](https://github.com/keplergl/kepler.gl/issues/351#issuecomment-548152534)) that multiple Redux versions result in Kepler being unable to connect to the `Provide`d store.
